### ISSUE TITLE
removeTxChannelInstance not called when deleting modulator causes segfault

### ISSUE THREAD

**Issue by f4exb** [f4exb/sdrangel#167]
This does not always occur but it does systematically in this case using the GUI:

- Instantiate a Tx device set with a LimeSDR transmiter
- start it
- add a SSB modulator
- delete the SSB modulator
- add a NFM modulator => segfault

Addind debug messages for registration ad removal of Tx channel instance the sequence show in the log is the following:
```
...
02.05.2018 23:59:26.245 (D) DeviceUISet::registerTxChannelInstance: sdrangel.channeltx.modssb 0x28880b8 0
02.05.2018 23:59:26.245 (D) DeviceUISet::renameTxChannelInstances: 1
...
02.05.2018 23:59:33.456 (D) DeviceUISet::registerTxChannelInstance: sdrangel.channeltx.modnfm 0x28ad248 1
02.05.2018 23:59:33.456 (D) DeviceUISet::renameTxChannelInstances: 2
Segmentation fault (core dumped)
```
As we can see the `removeTxChannelInstance` is not called when the SSB modulator is deleted. Therefore it is still present in the list as shown by the counter of instances (2) when invoking renameTxChannelInstances. The intermittent nature may suggest that removeTxChannelInstance is called in a different thread not synchronized with the one performing the registration.

**Comment by f4exb** [comment]
Located typo in SSBModGUI::~SSBModGUI(): using `removeRxChannelInstance` instead of `removeTxChannelInstance`

**Comment by f4exb** [comment]
Fix implemented in v3.14.5
